Implement constant time info retrieval (#344)
memgraph/documentation: pages/querying/schema.mdx

# Schema

Some queries are designed to retrieve information about the structure of the data within the database.
In the context of Memgraph, this refers to labels and properties associated with vertices, as well as edge types and properties associated with relationships.

## Schema associated queries

> <Callout type="info">

Reviewer: ```suggestion
```

> In order to use these queries you need to start Memgraph with the `--storage-enable-schema-metadata=True` option.

Reviewer: ```suggestion
To use schema-associated queries start Memgraph with the `--storage-enable-schema-metadata` [configuration flag](/configuration/configuration-settings#storage) set to `True`.
```

> </Callout>

Reviewer: ```suggestion
```

> {<h2><code> SHOW NODE_LABELS INFO </code> </h2>}

Reviewer: ```suggestion
## Information about the nodes

```

The `SHOW NODE_LABELS INFO` query provides a list of vertex-labels that currently exist or have existed at some point in the database.

```cypher
CREATE (:Dog {name: "Rex", age: 5})-[l:LOVES {how_much: "very"}]->(:Human:Owner {name: "Carl", age: 90});
```

> 

Reviewer: ```suggestion
The `SHOW NODE_LABELS INFO;` will return all the labels:
```

```cypher
SHOW NODE_LABELS INFO;
```

> 

Reviewer: ```suggestion
Result:

```

```plaintext
+-------------+
| node labels |
+-------------+
| "Owner"     |
| "Human"     |
| "Dog"       |
+-------------+
```

{<h2><code> SHOW EDGE_TYPES INFO </code> </h2>}
The `SHOW EDGE_TYPES INFO` query provides a list of edge-types that currently exist or have existed at some point in the database.

```cypher
CREATE (:Dog {name: "Rex", age: 5})-[l:LOVES {how_much: "very"}]->(:Human:Owner {name: "Carl", age: 90});
```

```cypher
SHOW EDGE_TYPESS INFO;
```

> 

Reviewer: ```suggestion
Result:

```

## Information about node labels

The `SHOW NODE_LABELS INFO;` query provides a list of node labels that currently exist in the database or have existed at some point.

If the database contains the following data:

```cypher
CREATE (:Dog {name: "Rex", age: 5})-[l:LOVES {how_much: "very"}]->(:Human:Owner {name: "Carl", age: 90});
```
The `SHOW NODE_LABELS INFO;` will return all the labels:
```cypher
SHOW NODE_LABELS INFO;
```
Result:

```plaintext
+-------------+
| node labels |
+-------------+
| "Owner"     |
| "Human"     |
| "Dog"       |
+-------------+
```

## Information about relationship types

The `SHOW EDGE_TYPES INFO;` query provides a list of relationship types that currently exist in the database or have existed at some point.

If the database contains the following data:

```cypher
CREATE (:Dog {name: "Rex", age: 5})-[l:LOVES {how_much: "very"}]->(:Human:Owner {name: "Carl", age: 90});
```
The `SHOW EDGE_TYPES INFO;` will return all the types:

```cypher
SHOW EDGE_TYPES INFO;
```
Result:

```plaintext
+------------+
| edge types |
+------------+
| "LOVES"    |
+------------+
```

> ## Schema associated CALL procedures

Reviewer: ```suggestion
## Schema associated procedures
```
